pkg/jobmanager/bundles.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+package jobmanager
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/facebookincubator/contest/pkg/job"
+	"github.com/facebookincubator/contest/pkg/pluginregistry"
+	"github.com/facebookincubator/contest/pkg/test"
+)
+
+// newReportingBundles returns the bundles for the run report and the final report
+func newReportingBundles(registry *pluginregistry.PluginRegistry, jobDescriptor *job.Descriptor) ([]*job.ReporterBundle, []*job.ReporterBundle, error) {
+
+	var (
+		runReporterBundles   []*job.ReporterBundle
+		finalReporterBundles []*job.ReporterBundle
+	)
+
+	for _, reporter := range jobDescriptor.Reporting.RunReporters {
+		if strings.TrimSpace(reporter.Name) == "" {
+			return nil, nil, errors.New("invalid empty or all-whitespace run reporter name")
+		}
+		bundle, err := registry.NewRunReporterBundle(reporter.Name, reporter.Parameters)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to create bundle for run reporter '%s': %v", reporter.Name, err)
+		}
+		runReporterBundles = append(runReporterBundles, bundle)
+	}
+
+	for _, reporter := range jobDescriptor.Reporting.FinalReporters {
+		if strings.TrimSpace(reporter.Name) == "" {
+			return nil, nil, errors.New("invalid empty or all-whitespace final reporter name")
+		}
+		bundle, err := registry.NewFinalReporterBundle(reporter.Name, reporter.Parameters)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to create bundle for final reporter '%s': %v", reporter.Name, err)
+		}
+		finalReporterBundles = append(finalReporterBundles, bundle)
+	}
+
+	return runReporterBundles, finalReporterBundles, nil
+}
+
+func newBundlesFromSteps(descriptors []test.StepDescriptor, registry *pluginregistry.PluginRegistry) ([]test.StepBundle, error) {
```

Comment:
Mh... the intention of this naming was to highlight that the bundles here are created from descriptors passed directly as part of the signature of the function, rather than fetched inside the function. I feel this information is lost with the naming suggested?